src/react/apiBase.ts

```diff
@@ -81,67 +52,21 @@ export function useStoreMapBase<State, Result, Keys extends ReadonlyArray<any>>(
   if (!is.store(store)) throwError('useStoreMap expects a store')
   if (!Array.isArray(keys)) throwError('useStoreMap expects an array as keys')
   if (typeof fn !== 'function') throwError('useStoreMap expects a function')
-  const result = React.useRef<{
-    fn: (state: State, keys: Keys) => Result
-    upd: (update: Result, current: Result) => boolean
-    val: Result
-    init: boolean
-    store: Store<State>
-    active: boolean
-  }>({} as any)
-  const refState = result.current
-  refState.fn = fn
-  refState.upd = updateFilter
-  refState.init = refState.store === store
-  refState.store = store
-  refState.active = true
-  useIsomorphicLayoutEffect(
-    () => () => {
-      if (result.current) {
-        result.current.active = false
-      }
-    },
-    [],
+
+  const subscribe = (cb: () => void) => createWatch(store, cb, scope)
+  const read = () => stateReader(store, scope)
+
+  const value = useSyncExternalStoreWithSelector(
+    subscribe,
+    read,
+    read,
+    state => fn(state, keys),
```

Comment:
`keys` array was used in `React.useMemo` in previous implementation to avoid unnecessary updates, but new official `useSyncExternalStoreWithSelector` does not accept any keys 😢 

Should we add some custom memoization here?

Reply:
Memoization is needed:
https://github.com/effector/effector/pull/655/commits/6790129f9fac579ce47dd870e703c30d846e4e59